### PR TITLE
Profiling and cleanup

### PR DESCRIPTION
**Pull request by tibo-mllr**
Add profling for all methods

### CODE REVIEW

**Comment by tibo-mllr** [comment]
Only time profiling for the moment, will try to add memory

**Comment by tibo-mllr** [comment]
Finally done, both time and input-varying profling/benchmark.

**Comment by LuukEbenau** [comment]
Looks good!

File: time_profiling.py
Comment (LuukEbenau) [inline time_profiling.py:1]: Seems professional!

File: torch_code/main.py
Comment (LuukEbenau) [inline torch_code/main.py:1]: Did you run a linter or something?

**Review by LuukEbenau — APPROVED** [review]

**Review by LuukEbenau — APPROVED** [review]

File: torch_code/main.py
Comment (tibo-mllr) [reply torch_code/main.py:1]: Yep, look at [this file](./.vscode/settings.json). Normally we all have, if we have the extension

**Pull request merged**